diff --git a/drivers/net/ethernet/mellanox/mlx5/core/port.c b/drivers/net/ethernet/mellanox/mlx5/core/port.c
index 367df52..e53ac79 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/port.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/port.c
@@ -313,7 +313,7 @@ static int mlx5_query_module_id(struct mlx5_core_dev *dev, int module_num,
 
 	status = MLX5_GET(mcia_reg, out, status);
 	if (status) {
-		mlx5_core_err(dev, "query_mcia_reg failed: status: 0x%x\n",
+		mlx5_core_dbg(dev, "query_mcia_reg failed: status: 0x%x\n",
 			      status);
 		return -EIO;
 	}
