From f78c3d7e62037b6f3964e67f4cff0abe39c57a81 Mon Sep 17 00:00:00 2001
From: Stepan Blyschak <stepanb@nvidia.com>
Date: Sat, 15 Oct 2022 11:36:02 +0000
Subject: [PATCH 1/2] [dplane_fpm_nl] Use VRF if_index instead of table_id in
 FPM messages

Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>
---
 zebra/rt_netlink.c | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/zebra/rt_netlink.c b/zebra/rt_netlink.c
index 70ff65606..771f17d38 100644
--- a/zebra/rt_netlink.c
+++ b/zebra/rt_netlink.c
@@ -1985,13 +1985,19 @@ ssize_t netlink_route_multipath_msg_encode(int cmd,
 #endif
 	/* Table corresponding to this route. */
 	table_id = dplane_ctx_get_table(ctx);
-	if (table_id < 256)
-		req->r.rtm_table = table_id;
-	else {
-		req->r.rtm_table = RT_TABLE_UNSPEC;
-		if (!nl_attr_put32(&req->n, datalen, RTA_TABLE, table_id))
-			return 0;
-	}
+
+    if (!fpm) {
+        if (table_id < 256)
+            req->r.rtm_table = table_id;
+        else {
+            req->r.rtm_table = RT_TABLE_UNSPEC;
+            if (!nl_attr_put32(&req->n, datalen, RTA_TABLE, table_id))
+                return 0;
+        }
+    } else {
+        /* Put vrf if_index instead of table id */
+        req->r.rtm_table = dplane_ctx_get_vrf(ctx);
+    }
 
 	if (IS_ZEBRA_DEBUG_KERNEL)
 		zlog_debug(
-- 
2.30.2

