From 46757392b4228b27658a79c6821370dfc421a380 Mon Sep 17 00:00:00 2001
From: Stepan Blyschak <stepanb@nvidia.com>
Date: Sun, 19 Mar 2023 22:46:15 +0200
Subject: [PATCH] remove fib install pending flag if new_select

Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>
---
 bgpd/bgp_route.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/bgpd/bgp_route.c b/bgpd/bgp_route.c
index d4b5066c6..7ccc557df 100644
--- a/bgpd/bgp_route.c
+++ b/bgpd/bgp_route.c
@@ -3118,6 +3118,9 @@ static void bgp_process_main_one(struct bgp *bgp, struct bgp_dest *dest,
 
 			bgp_zebra_announce(dest, p, new_select, bgp, afi, safi);
 		} else {
+			if (new_select)
+				UNSET_FLAG(dest->flags, BGP_NODE_FIB_INSTALL_PENDING);
+
 			/* Withdraw the route from the kernel. */
 			if (old_select && old_select->type == ZEBRA_ROUTE_BGP
 			    && (old_select->sub_type == BGP_ROUTE_NORMAL
-- 
2.17.1

