From 54fd83574159ed98f9bcb9c8a5171d91d85323ef Mon Sep 17 00:00:00 2001
From: stepanb <stepanb@contoso.com>
Date: Tue, 21 Mar 2023 21:43:21 +0000
Subject: [PATCH] [bgp] fix connected route not advertised

Signed-off-by: stepanb <stepanb@contoso.com>
---
 bgpd/bgp_route.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/bgpd/bgp_route.c b/bgpd/bgp_route.c
index d4b5066c6..c0004ac7e 100644
--- a/bgpd/bgp_route.c
+++ b/bgpd/bgp_route.c
@@ -3953,19 +3953,6 @@ int bgp_update(struct peer *peer, const struct prefix *p, uint32_t addpath_id,
 		goto filtered;
 	}
 
-	/* The flag BGP_NODE_FIB_INSTALL_PENDING is for the following
-	 * condition :
-	 * Suppress fib is enabled
-	 * BGP_OPT_NO_FIB is not enabled
-	 * Route type is BGP_ROUTE_NORMAL (peer learnt routes)
-	 * Route is being installed first time (BGP_NODE_FIB_INSTALLED not set)
-	 */
-	if (bgp_fibupd_safi(safi) && BGP_SUPPRESS_FIB_ENABLED(bgp)
-	    && (sub_type == BGP_ROUTE_NORMAL)
-	    && (!bgp_option_check(BGP_OPT_NO_FIB))
-	    && (!CHECK_FLAG(dest->flags, BGP_NODE_FIB_INSTALLED)))
-		SET_FLAG(dest->flags, BGP_NODE_FIB_INSTALL_PENDING);
-
 	/* If maximum prefix count is configured and current prefix
 	 * count exeed it.
 	 */
-- 
2.30.2

