diff --git a/debian/mlxbf-scripts.install b/debian/mlxbf-scripts.install
index 7ee446d..093e3c3 100644
--- a/debian/mlxbf-scripts.install
+++ b/debian/mlxbf-scripts.install
@@ -1,4 +1,3 @@
 bfacpievt bfbootmgr bfcfg bfcpu-freq bfdracut bffamily bfgrubcheck bfinst bfup usr/bin/
 bfpart bfpxe bfrec bfrshlog bfsbdump bfsbkeys bfsbverify bfvcheck bfver mlx-mkbfb bfhcafw usr/bin/
-bfvcheck.service bfup.service lib/systemd/system/
 mlx-uefi.quirk usr/share/fwupd/quirks.d/
diff --git a/debian/postinst b/debian/postinst
deleted file mode 100755
index 2b85abd..0000000
--- a/debian/postinst
+++ /dev/null
@@ -1,22 +0,0 @@
-#!/bin/sh -e
-OPTS=""
-
-if [ -n "$D" ]; then
-    OPTS="--root=$D"
-fi
-
-if which deb-systemd-helper >/dev/null 2>/dev/null; then
-	if [ -z "$D" ]; then
-		deb-systemd-helper daemon-reload
-	fi
-
-	deb-systemd-helper "$OPTS" enable bfvcheck.service
-	deb-systemd-helper "$OPTS" enable bfup.service
-
-	if [ -z "$D" ]; then
-		deb-systemd-helper restart bfvcheck.service
-		deb-systemd-helper restart bfup.service
-	fi
-fi
-
-#DEBHELPER#
