--- a/patch/series
+++ b/patch/series
@@ -108,6 +108,9 @@
 0045-i2c-mlxcpld-Fix-criteria-for-frequency-setting.patch
 0046-i2c-mlxcpld-Reduce-polling-time-for-performance-impr.patch
 0047-i2c-mlxcpld-Allow-flexible-polling-time-setting-for-.patch
+0049-leds-mlxreg-Provide-conversion-for-hardware-LED-colo.patch
+0050-leds-mlxreg-Skip-setting-LED-color-during-initializa.patch
+0051-leds-mlxreg-Allow-multi-instantiation-of-same-name-L.patch
 0053-mlxsw-core-Avoid-creation-virtual-hwmon-objects-by-t.patch
 0054-mlxsw-minimal-Simplify-method-of-modules-number-dete.patch
 0055-platform_data-mlxreg-Add-new-type-to-support-modular.patch
@@ -220,15 +223,20 @@
 0164-hwmon-jc42-Add-support-for-Seiko-Instruments-S-34TS0.patch
 0165-platform-mellanox-mlxreg-io-Add-locking-for-io-opera.patch
 0166-DS-leds-leds-mlxreg-Send-udev-event-from-leds-mlxreg.patch
+0167-DS-lan743x-Add-support-for-fixed-phy.patch
+0168-TMP-mlxsw-minimal-Ignore-error-reading-SPAD-register.patch
+0169-TMP-mlxsw-i2c-Prevent-transaction-execution-for-spec.patch
 0170-i2c-mlxcpld-Fix-register-setting-for-400KHz-frequenc.patch
 0171-platform-mellanox-mlxreg-lc-Fix-cleanup-on-failure-a.patch
 0172-DS-platform-mlx-platform-Add-SPI-path-for-rack-switc.patch
 0173-mlxsw-core-Add-support-for-OSFP-transceiver-modules.patch
+0174-DS-mlxsw-core_linecards-Skip-devlink-and-provisionin.patch
 0175-hwmon-pmbus-Add-support-for-Infineon-Digital-Multi-p.patch
 0176-platform-mellanox-fix-reset_pwr_converter_fail-attri.patch
 0177-Documentation-ABI-fix-description-of-fix-reset_pwr_c.patch
 0178-platform-mellanox-Introduce-support-for-next-generat.patch
 0180-hwmon-pmbus-Fix-sensors-readouts-for-MPS-Multi-phase.patch
+0181-Revert-Fix-out-of-bounds-memory-accesses-in-thermal.patch
 0182-platform-mellanox-Introduce-support-of-new-Nvidia-L1.patch
 0183-platform-mellanox-Split-initialization-procedure.patch
 0184-platform-mellanox-Split-logic-in-init-and-exit-flow.patch
@@ -316,6 +324,7 @@
 0265-hwmon-mlxreg-fan-Return-zero-speed-for-broken-fan.patch
 0266-UBUNTU-SAUCE-mlxbf-pmc-Bug-fix-for-BlueField-3-count.patch
 0267-UBUNTU-SAUCE-mmc-sdhci-of-dwcmshc-add-the-missing-de.patch
+0268-DS-mlxsw-core_linecards-Disable-firmware-bundling-ma.patch
 0269-platform-mellanox-Cosmetic-changes.patch
 0270-platform-mellanox-Fix-order-in-exit-flow.patch
 0271-platform-mellanox-Add-new-attributes.patch
@@ -331,6 +340,7 @@
 0281-platform-mellanox-mlx-platform-Modify-health-and-pow.patch
 0282-platform-mellanox-mlx-platform-add-support-of-5th-CP.patch
 0283-mlxsw-core_hwmon-Align-modules-label-name-assignment.patch
+9002-TMP-fix-for-fan-minimum-speed.patch
 ###-> mellanox_hw_mgmt-end
 
 # Cisco patches for 5.10 kernel
